### PR TITLE
fix(external docs): Render encoding as required when codecs are enabled

### PR DESCRIPTION
**Pull request by spencergilbert**
- chore: Update hugo config to allow babel usage
- fix(external docs): Render encoding as required when codec is enabled

Theoretically I think this should be derived from when `codec.enabled == true` but I wasn't sure how to accomplish that. This only partially closes https://github.com/vectordotdev/vector/issues/10051 as we still don't render required children objects (`codec`) in this case as required in the final HTML/CSS

### CODE REVIEW

File: website/cue/reference/components/sinks.cue [hunk 1]
```diff
@@ -176,8 +176,8 @@ components: sinks: [Name=string]: {
 			if features.send.encoding.enabled {
 				encoding: {
 					description: "Configures the encoding specific sink behavior."
-					required:    false
-					common:      true
+					required:    features.send.encoding.required
+					if !features.send.encoding.required {common: true}
```
Comment (jszwedko) [inline website/cue/reference/components/sinks.cue:180]: Probably needs a `cue fmt`?

**Review by jszwedko — COMMENTED** [review]
Apologies for not realizing this sooner, but I think the top-level `required` field you've added could actually be derived from `features.send.encoding.codec.enabled`. If `features.send.encoding.codec.enabled` is `false` then `encoding` is not required, if `features.send.encoding.codec.enabled` is `true` then `encoding` is required. That'd save the addition of the `required` field you added here.

**Comment by spencergilbert** [comment]
> Apologies for not realizing this sooner, but I think the top-level `required` field you've added could actually be derived from `features.send.encoding.codec.enabled`. If `features.send.encoding.codec.enabled` is `false` then `encoding` is not required, if `features.send.encoding.codec.enabled` is `true` then `encoding is required. That'd save the addition of the `required` field you added here.

I didn't realize you could "invert" the relationship and set parent fields based on child objects, I'll give it a try...

File: website/config.toml [hunk 2]
```diff
@@ -312,3 +312,16 @@ weight = 1
 name = "Privacy policy"
 url = "https://github.com/vectordotdev/vector/blob/master/PRIVACY.md"
 weight = 2
+
+[security]
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply website/config.toml:316]: 👍 wasn't entirely sure it was necessary/functioning in our actual deploy but I needed it for hugo to work locally

**Review by jszwedko — APPROVED** [review]
🎉

**Comment by jszwedko** [comment]
Merging since the required checks would never pass here.

**Pull request merged**